### ISSUE TITLE
Facebook plugin : showDialog apprequests new issue

### ISSUE THREAD

**Issue by JerryBels** [Wizcorp/phonegap-facebook-plugin#908]
Hello guys,

I installed last month the [Facebook plugin](https://github.com/Wizcorp/phonegap-facebook-plugin) for Cordova. Last week I implemented the [showDialog](https://github.com/Wizcorp/phonegap-facebook-plugin#show-a-dialog) to let people invite their friends to the app using "apprequests" method. It worked when I tried with two friends, they got the invites.

Today, trying again, everything works fine - except that the people I invite never get any notification.

Here is the code of the call (using [NGCordova's wrapper](http://ngcordova.com/docs/plugins/facebook/)) :

```
$cordovaFacebook.showDialog({ method: "apprequests", message: "Come on man, check out my application." })
.then(function(success) {
    console.log("success", success);
    // success
}, function (error) {
    console.log("error", error);
    // error
});
```

The console shows success : 

```
success Object {to: Array[3], request: "id_of_request"}
    request: "id_of_request"
    to: Array[3]
        0: "user_id_1"
        1: "user_id_2"
        2: "user_id_3"
        length: 3
```

The user ids were good also. But no notifications for them on their Facebook account.

What I tried so far : 
- Reinstalling Facebook plugin
- Create another Facebook app and use its credentials instead
- Installing the latest version of NGCordova (unlikely to help, is simply a wrapper...)
- Using other methods, like "feed", and post to my wall - it works.

Taking NGCordova out of the picture, using basic function behave the same : 

```
facebookConnectPlugin.showDialog({ method: "apprequests", message: "Come on man, check out my application." }, function(success) {
        console.log(success);
    }, function(error) {
        console.log(error);
    });
```

This triggers success as well and still no notifications.

Thanks ahead for any help, I think I'm out of ideas...

**Comment by bastiRe** [comment]
Hey,
I have the same issue. facebookConnectPlugin.showDialog works perfectly but doesn't actually create a notification. Did you figure out what the problem was @JerryBels ?

**Comment by JerryBels** [comment]
No, nothing :( I assumed that's because of the Facebook app, which is a dev version that hasn't been validated, and I will review this more later...

**Comment by bastiRe** [comment]
So what was the difference between the time it worked and the time it didn't?

**Comment by JerryBels** [comment]
Not the slightest hint :(

**Comment by bastiRe** [comment]
I figured it out for me. You need to have Canvas as one of the registered platforms so that people receive Notifications on the web version of Facebook. Same goes for the other platforms.

**Comment by JerryBels** [comment]
Canvas as a registered platform ? What does it mean ?

**Comment by bastiRe** [comment]
You can add different platforms in the settings of your Facebook app. One of them is Facebook Canvas. To have notifications delivered on the Desktop version of Facebook you need to add Canvas as a platform.

**Comment by JerryBels** [comment]
But this platform doesn't seems related to the notifications to me ? It asks me for a https page to display inside the canvas, which I currently don't have, also !

**Comment by antigirl** [comment]
seems to be a bug in the api. not plugins fault :/
https://developers.facebook.com/bugs/532388000129311

**Comment by antigirl** [comment]
btw i've found this http://stackoverflow.com/questions/16792782/app-requests-successful-but-no-notification-shown, 2nd response reads 'The second one - make sure your app is NOT in SandBox Mode. If it is it won't send any notifications. I lost hours before figuring that out and changing the code in all possible ways.'

that might be my problem

**Comment by JerryBels** [comment]
@antigirl that what I assumed also, that's why I'm not waisting too much time on this right now.

**Comment by antigirl** [comment]
@bastiRe @JerryBels  figured it out, it because the app is in sandbox/test mode it doesnt send a notification. You can add testers to the app. [https://developers.facebook.com/docs/apps/test-users#managetool]  I tried it on my testers and it seems to be sending a notification :)
So assuming once app is live, it will send notifications

**Comment by nehatha** [comment]
I am still facing the same issue even after adding the tester.Did any one get any other solution?

**Comment by MoemenMostafa** [comment]
This issue happens because this plugin is using Facebook SDK v3 and the current minimum SDK is v4 that is why you will never get a push notification (you need to use appInvite instead). You can use the following fork to fix this problem (https://github.com/jeduan/cordova-plugin-facebook4)

**Comment by aboutsam** [comment]
hello moemen,
if i use appinvite, i not get the same result of the request. and if i use "showDialog" i get the information my app i not a game.. do you know how to get the same result of "showdialog"?

**Comment by MoemenMostafa** [comment]
@aboutsam what you are trying to achieve here?
